code-challenges.test.js

```diff
@@ -12,40 +12,125 @@
+
+  
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 
 // b) Create the function that makes the test pass.
 
+//Create a function called data that takes in an array of objects
+  //Map through the array
+    //for each object in the array... return string interpolation with the value of the name key with proper capitalization of the first letters of the first and last names, the word 'is', the value associated with the occupation key, and a period
+    
+
+
+
+
+const data = (array) => {
+   return array.map(object => {
+    return `${object.name.split(" ").map(value => {return value.charAt(0).toUpperCase() + value.slice(1)}).join(" ")} is ${object.occupation}.`
+  })
+}
+
+
+          // PASS  ./code-challenges.test.js
+          // data
+          //   ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (3 ms)
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe("modulo3", () => { 
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+  // Expected output: [ 2, 0, -1, 0 ]
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+  // Expected output: [ 2, 1, -1 ]
+  
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () =>{
+  expect(modulo3(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+  expect(modulo3(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+
+          // FAIL  ./code-challenges.test.js
+          // modulo3
+          //   ✕ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (1 ms)
+          // ● modulo3 › takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
+          //   ReferenceError: modulo3 is not defined
+
+
 
 
 // b) Create the function that makes the test pass.
 
+//Create a function called modulo3 that takes in an array of mixed data types
+  //filter out only the values that have the "number" datatype
+  //return a map through the result of the filtering and apply a modulo of three to each of the values
+
+  const modulo3 = (array) => {
+    return array.filter(value => {return typeof(value)==="number"}).map(value => {return value % 3})
+  }
 
```

Comment:
Great working to a single line return statement. Arrow functions allow for a shorter syntax if it is a single statement. For instance, you can remove return and the curly braces as a refactor
  `const modulo3 = (array) => {
    return array.filter(value => typeof(value)==="number").map(value => value % 3)
  }`

Just wanted to share since you are doing such great work reducing your lines